openquake/hazardlib/gsim/campbell_1997.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+#
+# Copyright (C) 2013-2018 GEM Foundation
+#
+# OpenQuake is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module exports :class:`Campbell1997`
+"""
+import numpy as np
+
+from openquake.hazardlib.gsim.base import GMPE
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA
+
+
+class Campbell1997(GMPE):
+    """
+    Implements GMPE (PGA) by Campbell, Kenneth W. "Empirical near-source
+    attenuation relationships for horizontal and vertical components of peak
+    ground acceleration, peak ground velocity, and pseudo-absolute acceleration
+    response spectra." Seismological research letters 68.1 (1997): 154-179.
+    """
+    #: Supported TRT active...we specify active_shallow_crust
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.ACTIVE_SHALLOW_CRUST
+
+    #: Supported intensity measure types are PGA, PGV, PSA, but we only define
+    #: PGA because this is the only IMT used by an implemented model (09/18)
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([
+        PGA
+    ])
+
+    #: Supported intensity measure component is the horizontal component
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.AVERAGE_HORIZONTAL
+
+    #: Supported standard deviation type is only total, see equation 4, pg 164
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL,
+    ])
+
+    #: Requires vs30
+    REQUIRES_SITES_PARAMETERS = set(('vs30',))
+
+    #: Required rupture parameters are magnitude and top of rupture depth
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag', 'rake'))
+
+    #: Required distance measure is closest distance to rupture
```

Comment:
After quickly scanning the publication, I see they are actually using Rseis (distance to seismogenic rupture ). Might be worth mentioning that we using Rrup instead, which should be very similar to Rseis. Otherwise LGTM